test/common/run-tests

```diff
@@ -165,7 +165,7 @@ class Test:
 
         # be quiet in TAP mode for successful tests
         lines = self.output.strip().splitlines(keepends=True)
-        if print_tap and self.returncode == 0 and len(lines) > 0:
+        if print_tap and self.returncode == 0 and len(lines) > 0 and b'# WARNING: expectedFailure' not in self.output:
```

Comment:
I was wondering if we could get some orange or purple or something for these on the logs page...

Reply:
Yes, that'd be nice -- we need to do that in logs.html indeed, as a new recognized state.

Reply:
See the combined efforts of #18045 and https://github.com/cockpit-project/bots/pull/4180 , this happens now.